types/wlr_scene.c

```diff
@@ -0,0 +1,309 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <wlr/backend.h>
+#include <wlr/render/wlr_renderer.h>
+#include <wlr/types/wlr_matrix.h>
+#include <wlr/types/wlr_scene.h>
+#include <wlr/types/wlr_surface.h>
+#include "util/signal.h"
+
+static struct wlr_scene *scene_node_get_root(struct wlr_scene_node *node) {
```

Comment:
nit: this reads like it'd walk up the node tree until it found the root node. Perhaps better to call it `scene_from_root_node` or something.